src/controllers/prove.ts

```diff
@@ -1,16 +1,24 @@
+import * as JSONbig from 'json-bigint'
 import { Body, Controller, Ctx, Get, Params, Post } from 'amala'
 import { Context } from 'koa'
 import { Job, JobModel } from '@/models/Job'
 import { notFound } from '@hapi/boom'
-import Inputs from '@/validators/Inputs'
 import JobStatus from '@/models/JobStatus'
+import JsonProofInput from '@/validators/JsonProofInput'
+import ProofInput from '@/validators/ProofInput'
 import ProofResultParams from '@/validators/ProofResultParam'
 
 @Controller('/')
 export default class ProveController {
   @Post('/')
-  async proof(@Body({ required: true }) input: Inputs) {
-    const job = await JobModel.create({ input })
+  async proof(@Body({ required: true }) input: JsonProofInput) {
+    const { TPreComputes, U, s } = input
+    const parsedInput: ProofInput = {
```

Comment:
Do you need to explicitly specify a type here? I think the `parse` should return a proper type, if it doesn't, than this is fine

Reply:
Without the `ProofInput` type, the const will have any inside object: 
```
{
  TPreComputes: any
  U: any
  s: any
}
```